test/html.test.js

```diff
@@ -665,3 +672,88 @@ QUnit.test('does not wrap mp4 source buffers', function() {
   );
   QUnit.equal(mediaSource.sourceBuffers.length, 2, 'created native buffers');
 });
+
+QUnit.test('can get activeSourceBuffers', () => {
+  let mediaSource = new videojs.MediaSource();
+
+  QUnit.ok(Array.isArray(mediaSource.activeSourceBuffers));
+});
+
+QUnit.test('active source buffers are updated on each buffer\'s first updateend', () => {
+  let mediaSource = new videojs.MediaSource();
+  let updateCallCount = 0;
+  let sourceBuffer;
+
+  mediaSource.updateActiveSourceBuffers_ = () => {
+    updateCallCount++;
+  };
+
+  sourceBuffer = mediaSource.addSourceBuffer('video/mp2t');
+  initializeNativeSourceBuffers(sourceBuffer);
+
+  QUnit.equal(updateCallCount, 0,
+              'active source buffers not updated on adding source buffer');
+
+  sourceBuffer.videoBuffer_.trigger('updateend');
+  QUnit.equal(updateCallCount, 1,
+              'active source buffers updated after first updateend');
+
+  sourceBuffer.videoBuffer_.trigger('updateend');
+  QUnit.equal(updateCallCount, 1,
+              'active source buffers not updated on second updateend');
+
+  sourceBuffer = mediaSource.addSourceBuffer('video/mp2t');
+  initializeNativeSourceBuffers(sourceBuffer);
+
+  QUnit.equal(updateCallCount, 1,
+              'active source buffers not updated on adding second source buffer');
+
+  sourceBuffer.videoBuffer_.trigger('updateend');
+  QUnit.equal(updateCallCount, 2,
+              'active source buffers updated after first updateend of new source buffer');
+
+  sourceBuffer.videoBuffer_.trigger('updateend');
+  QUnit.equal(updateCallCount, 2,
+              'active source buffers not updated after second updateend of new source ' +
+              'buffer');
+});
+
+// running under the assumption that we will have a max of two source buffers
```

Comment:
I don't see any assumptions in this test that strike me as untrue if we allowed more than two source buffers (given the test only creates two). Am I missing something?